R/Connection.R

```diff
@@ -191,6 +195,184 @@ setMethod(
+#' for further details.
+#'
+#' @return data.frame with columns
+#' - table_catalog
+#' - table_schema
+#' - table_name
+#' - table_remarks
+setGeneric(
+  "odbcConnectionTables",
+  valueClass = "data.frame",
+  function(conn, name, ...) {
+    standardGeneric("odbcConnectionTables")
+  }
+)
+
+#' @rdname odbcConnectionTables
+#' @param table_type List tables of this type, for example 'VIEW'.
+#' See odbcConnectionTableTypes for a listing of available table
+#' types for your connection.
+setMethod(
+  "odbcConnectionTables",
+  c("OdbcConnection", "Id"),
+  function(conn, name, table_type = NULL) {
+
+    odbcConnectionTables(conn,
+      name = id_field(name, "table"),
+      catalog_name = id_field(name, "catalog"),
+      schema_name = id_field(name, "schema"),
+      table_type = table_type)
+  }
+)
+
+#' @rdname odbcConnectionTables
+#' @param catalog_name character catalog where we wish to query for
+#' available tables
+#' @param schema_name character schema where we wish to query for
+#' available tables.
+setMethod(
+  "odbcConnectionTables",
+  c("OdbcConnection", "character"),
+  function(conn, name, catalog_name = NULL, schema_name = NULL, table_type = NULL) {
+
+    connection_sql_tables(conn@ptr,
+      catalog_name = catalog_name,
+      schema_name = schema_name,
+      table_name = name,
+      table_type = table_type)
+
+  }
+)
+
+setMethod(
+  "odbcConnectionTables",
+  c("OdbcConnection", "missingOrNULL"),
+  function(conn, name, catalog_name = NULL, schema_name = NULL, table_type = NULL) {
+
+    odbcConnectionTables(conn,
+      name = "%",
+      catalog_name = catalog_name,
+      schema_name = schema_name,
+      table_type = table_type)
+
+  }
+)
+
+#' @rdname odbcConnectionTables
+setMethod(
+  "odbcConnectionTables", c("OdbcConnection", "SQL"),
+  function(conn, name, table_type = NULL) {
+    odbcConnectionTables(conn, dbUnquoteIdentifier(conn, name)[[1]], ...)
+  })
+
+#' odbcConnectionCatalogs
+#'
+#' This function returns a listing of available
+#' catalogs.
+#' @param conn OdbcConnection
+#' @rdname odbcConnectionCatalogs
+setGeneric(
+  "odbcConnectionCatalogs",
+  valueClass = "character",
+  function(conn) {
+    standardGeneric("odbcConnectionCatalogs")
+  }
+)
+
+#' @rdname odbcConnectionCatalogs
+setMethod(
+  "odbcConnectionCatalogs",
+  c("OdbcConnection"),
+  function(conn) {
+    connection_sql_catalogs(conn@ptr)
+  }
+)
+
+#' odbcConnectionSchemas
+#'
+#' This function returns a listing of available
+#' schemas.
+#'
+#' Currently, for a generic connection the
+#' catalog_name argument is ignored.
+#'
+#' @param conn OdbcConnection
+#' @param catalog_name Catalog where
+#' we are looking to list schemas.
+#' @rdname odbcConnectionSchemas
+setGeneric(
+  "odbcConnectionSchemas",
+  valueClass = "character",
+  function(conn, catalog_name) {
+    standardGeneric("odbcConnectionSchemas")
+  }
```

Comment:
```suggestion
  function(conn, catalog_name = NULL) {
    standardGeneric("odbcConnectionSchemas")
  }
```

Does that allow you to avoid the new `missingOrNULL` class union?

Reply:
Good idea - done / thanks!